src/dash-table/derived/cell/operations.tsx

```diff
@@ -2,31 +2,38 @@ import * as R from 'ramda';
 import React from 'react';
 
 import { memoizeOneFactory } from 'core/memoizer';
+import { clearSelection } from 'dash-table/utils/actions';
 
 import {
     Data,
     Datum,
     SetProps,
     RowSelection,
-    ActiveCell,
     Indices
 } from 'dash-table/components/Table/props';
 
-function deleteRow(idx: number, activeCell: ActiveCell, data: Data, selectedRows: number[]) {
+function deleteRow(idx: number, data: Data, selectedRows: number[]) {
     const newProps: any = {
-        data: R.remove(idx, 1, data)
+        data: R.remove(idx, 1, data),
+        // We could try to adjust selection, but there are lots of edge cases
+        ...clearSelection
     };
-    if (R.is(Array, activeCell) && activeCell[0] === idx) {
-        newProps.active_cell = [];
-    }
 
-    if (R.is(Array, selectedRows) && R.contains(idx, selectedRows)) {
-        newProps.selected_rows = R.without([idx], selectedRows);
+    if (R.is(Array, selectedRows) && R.any(i => i >= idx, selectedRows)) {
+        newProps.selected_rows = R.map(
+            // all rows past idx have now lost one from their index
+            i => i > idx ? i - 1 : i,
+            R.without([idx], selectedRows)
```

Comment:
This was a bug previously - delete a row and any selected rows after it kept the same index, not the same row contents.